stravalib/model.py

```diff
@@ -116,6 +125,80 @@ def __repr__(self):
         )
 
 
+class DeprecatedSerializableMixin(BaseModel):
+    """
+    Provides backward compatibility with legacy BaseEntity
+    """
+
+    @classmethod
+    def deserialize(cls, v):
```

Comment:
This will allow continued use of the legacy (de-)serialization methods, but it will raise a `DeprecationWarning`

Reply:
You can use (e.g.) ```Club.deserialize(some_dict)```, which will give you a Club object. That's all to it.

Reply:
oooh ok i understand now. the previous implementation was

`return model.Athlete.deserialize(raw, bind_client=self)` and it's now changed to `parse_obj(raw)`

i understand. i really don't see any issues with this approach. and am impressed that we can make such a big change to the API and still retain that backwards compatibility. I think you mentioned this early on in terms of raising warnings and such for the legacy methods. so essentially that is then what is happening here:

https://github.com/stravalib/stravalib/pull/315/files#diff-3f71efb739eb651ccb258a4bd554a784752bda3cd6c93cf9f01ddc0cdc1f5a36R139 

and we just need to develop a timeline for when we might drop legacy support? does that sound right? this all is looking great to me so far.